nats.go

```diff
@@ -2431,8 +2431,15 @@ func (nc *Conn) processInfo(info string) error {
 		}
 		nc.addURLToPool(fmt.Sprintf("%s://%s", nc.connScheme(), curl), true, saveTLS)
 	}
-	if hasNew && !nc.initc && nc.Opts.DiscoveredServersCB != nil {
-		nc.ach.push(func() { nc.Opts.DiscoveredServersCB(nc) })
+	if hasNew {
+		// If randomization is allowed, randomize the pool now that we have
+		// added all new URLs.
+		if !nc.Opts.NoRandomize {
+			nc.shufflePool()
```

Comment:
What happens to the server we are currently connected to?
Is it possible that this reshuffle, on next disconnect, causes us to try connecting from the server just disconnected?

Reply:
That means I guess that we should shuffle but only [1:] and leave at pos 0 the one we are currently connected to?

Reply:
I think so.. We should never try to reconnect to the server we were connected to unless its the only one.

Reply:
@derekcollison @matthiashanel Have another look please.